### DISCUSSION TITLE
String for AOVS

### DISCUSSION THREAD

**Discussion by RobinBruneau** [mitsuba-renderer/mitsuba3#398] [Q&A] [answered]
Hi,

I would like to use the AOV integrator, but I didn't find in the documentation the real string names for the differents options.
![image](https://user-images.githubusercontent.com/50364628/203017344-4a5a1426-afc4-4215-b9e1-6ad7ac70c40a.png)

In this example, **depth** is **dd.y:depth**, so I would have been nice to put the full string instead of just "depth" in the documentation.
All this to say that I would like to use other types, but I can't because I have no ideas of what subtring that I am supposed to add.

Could you tell me the real string for each types ?
And even better, could you update your documentation for other people ?

Thanks,
Robin

**Comment by njroussel** [comment, marked as answer]
Hi @RobinBruneau ,

I'm not quite sure I understand what you're trying to point out. The `aovs` field has to be filled with a list of `<name>:<type>` pairs, where the `type` is one of the values listed in your screenshot. As for the `name`, this is something the user can define and is not a restriction on Mitsuba's end. This `name` is then used in the `Bitmap` type and will also be used when writing to an `.exr`.
Back to the example in the plugin reference, the `dd.y` is just an abitrary choice for the purpose of the example but you, as a user, could set it to whatever you want.


In the [denoising how-to-guide](https://mitsuba.readthedocs.io/en/latest/src/how_to_guides/image_io_and_manipulation.html#Denoising), we make use of an `aov` integrator. Hopefully this might clarify a few things too.

**Reply by RobinBruneau** [reply]
Thanks, I just didn't understand that we could choose a random `name` . I was looking at a `name enumeration` somewhere that I couldn't find.
